### DISCUSSION TITLE
Error in chapter 4.

### DISCUSSION THREAD

**Discussion by iouzzrj** [IntelliTect/EssentialCSharp.Web#349] [Q&A] [answered]
```
The second method uses the define command line. Output 4.27 demonstrates this with Dotnet command-line interface.

Output 4.27
>dotnet.exe -define:CSHARP2PLUS TicTacToe.cs
```
I think that `dotnet TicTacToe.cs` cannot be used to compile the .cs source code. I see that `-define:CSHARP2PLUS` is the csc.exe syntax. So maybe you are referring to `csc -define:CSHARP2PLUS TicTacToe.cs`?

So, for dotnet.exe, I think I can use `dotnet build -p:DefineConstants=CSHARP2PLUS`?

**Comment by BenjaminMichaelis** [comment, marked as answer]
@iouzzrj Thank you for your feedback! 

When this content was originally written, we utilized csc. However, since transitioning to .NET, this issue has not resurfaced. I’ve made the necessary updates to the text, and you can expect to see the revised version live on our website in the coming days.

Regarding Output 4.27, we’ve made the following adjustment:

`dotnet build -p:DefineConstants=CSHARP2PLUS`

This command should be executed from the chapter 4 directory.

If you have any further questions or need additional assistance, feel free to reach out. We appreciate your engagement with our content! 😊

**Comment by BenjaminMichaelis** [comment]
Solved in https://github.com/IntelliTect/EssentialCSharp/issues/627
